lighthouse-core/gather/gatherers/dobetterweb/domnodes.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Gathers all DOM nodes on the page.
+ */
+
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+class AllDOMNodes extends Gatherer {
+  /**
+   * @param {!Object} options
+   * @return {!Promise<!Array<!Object>>}
+   */
+  afterPass(options) {
+    return options.driver.querySelectorAll('body, body /deep/ *');
```

Comment:
> We could get trickier and create a shared driver method that returns a promise and caches the result after the first call. Thoughts?

We probably should measure, I guess :) Maybe it's not so bad.

Beyond perf, though, for this gatherer there doesn't seem to be any benefit to keeping the array around